presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt

```diff
@@ -107,28 +107,74 @@ class QrActionProcessor(private val scanner: QrScannerActivity,
+                            }
+                        } else {
+                            ScanVoucherBaseErrorDialog("", scanner, reactivateDecoding).show()
+                        }
+                    }
+                    .onErrorReturn {
+                        var canOnResultVoucherScanned = true
+                        val error: Throwable = it
+                        if (error is RetrofitException && error.kind == RetrofitException.Kind.HTTP) {
 
-                    } else {
-                        val isAvailableScannedVoucher = !(it.voucher.isProduct != true && (
-                                it.voucher.amount
-                                        ?: 0.toBigDecimal()).compareTo(BigDecimal.ZERO) == 0)
-                        onResultVoucherScanned(address, isAvailableScannedVoucher)
+                            try {
+                                val newRecordError = retrofitExceptionMapper.mapToBaseApiError(error)
+
+                                if (error.responseCode == 403) {
+                                    canOnResultVoucherScanned = false
+                                    val message = if (newRecordError.message == null) "" else newRecordError.message
+                                    ScanVoucherBaseErrorDialog(message, scanner, reactivateDecoding).show()
+                                }
+                            } catch (e: Exception) {
+                            }
+                        }
 
+                        if (canOnResultVoucherScanned) {
+
+                            if (scanner.hasWindowFocus()) {
+                                onResultVoucherScanned(address, false)
+                            }
+                        }
                     }
-                }
-                .onErrorReturn {
+                    .subscribe()
+        } else {
+            vouchersRepository.getVoucherAsProvider(address)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .map {
+                        if (it.voucher.isProduct != true && it.allowedOrganizations.isEmpty()) {
+                            if (scanner.hasWindowFocus())
+                                ScanVoucherNotEligibleDialog(scanner, reactivateDecoding).show()
+
+                        } else {
+                            val isAvailableScannedVoucher = !(it.voucher.isProduct != true && (
+                                    it.voucher.amount
+                                            ?: 0.toBigDecimal()).compareTo(BigDecimal.ZERO) == 0)
+                            onResultVoucherScanned(address, isAvailableScannedVoucher)
 
-                    var canOnResultVoucherScanned = true
+                        }
+                    }
+                    .onErrorReturn {
+
+                        var canOnResultVoucherScanned = true
+
+                        Log.e("QR_ACTION", "scan voucher_error", it)
```

Comment:
can be removed?

Reply:
Only this line can be removed:
 Log.e("QR_ACTION", "scan voucher_error", it)